mayastor/src/nats.rs

```diff
@@ -0,0 +1,284 @@
+//! NATS message bus connecting mayastor to control plane (moac).
+//!
+//! It is designed to make sending events to control plane easy in the future.
+//! That's the reason for global sender protected by the mutex, that normally
+//! would not be needed and currently is used only to terminate the message bus.
+
+use futures::{channel::mpsc, select, FutureExt, StreamExt};
+use once_cell::sync::Lazy;
+use serde::{Deserialize, Serialize};
+use snafu::{ResultExt, Snafu};
+use std::{
+    env,
+    io::Error as IoError,
+    net::SocketAddr,
+    str::FromStr,
+    sync::Mutex,
+    time::Duration,
+};
+use tokio::{net::lookup_host, time::delay_for};
+use tokio_nats::{
+    connect,
+    Error as TokioNatsError,
+    NatsClient,
+    NatsConfigBuilder,
+};
+
+/// Mayastor sends registration messages in this interval (kind of heart-beat)
+const HB_INTERVAL: u64 = 10;
```

Comment:
what does this interval represent: s, m, h? Could we define it as a time::Duration?

Reply:
Good point! I changed the code.